### PR TITLE
Clean up handling of synchronous managers

### PR DESCRIPTION
**Pull request by blink1073**
Make the async contents manager the default, and add deprecation warning when using synchronous contents or kernel managers.

### CODE REVIEW

File: jupyter_server/serverapp.py [hunk 1]
```diff
@@ -1435,38 +1433,13 @@ def template_file_path(self):
         help="""If True, display controls to shut down the Jupyter server, such as menu items or buttons.""",
     )
 
-    # REMOVE in VERSION 2.0
```
Comment (blink1073) [inline jupyter_server/serverapp.py:1438]: @Zsailer I believe removing this is what is causing the nbclassic downstream test to fail.  Do we want to keep this in for another major release?

Comment (blink1073) [reply jupyter_server/serverapp.py:1438]: Okay, punted to 3.0

Comment (blink1073) [reply jupyter_server/serverapp.py:1438]: Okay, so the actual issue occurs when we change the default value from `LargeFileManager` -> `AsyncLargeFileManager`, which was the main idea of this change.  😅

Comment (blink1073) [reply jupyter_server/serverapp.py:1438]: Ah, nbclassic was assuming synchronous contents manager:

```
  tests/test_notebookapp.py::test_tree_handler
    /home/runner/test_venv/downstream_test/nbclassic/nbclassic/tree/handlers.py:53: RuntimeWarning: coroutine 'AsyncFileContentsManager.dir_exists' was never awaited
      if cm.dir_exists(path=path):
    Enable tracemalloc to get traceback where the object was allocated.
    See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
  
  tests/test_notebookapp.py::test_tree_handler
    /home/runner/test_venv/downstream_test/nbclassic/nbclassic/tree/handlers.py:54: RuntimeWarning: coroutine 'AsyncFileContentsManager.is_hidden' was never awaited
```

We need to update it regardless.

**Comment by blink1073** [comment]
All green!

**Review by Zsailer — APPROVED** [review]
Thanks, @blink1073! Great stuff here!

**Pull request merged**